packages/css/src/components/button/button.scss

```diff
@@ -95,3 +95,17 @@
     color: var(--ams-button-tertiary-hover-color);
   }
 }
+
+.ams-button--icon-only {
+  padding-block: var(--ams-button-icon-only-padding-block);
+  padding-inline: var(--ams-button-icon-only-padding-inline);
+}
+
+// TODO: Proposition
+// .ams-button--icon-start {
+//   padding-inline-start: var(--ams-button-icon-start-padding-inline-start);
+// }
+
+// .ams-button--icon-end {
+//   padding-inline-end: var(--ams-button-icon-end-padding-inline-end);
+// }
```

Comment:
The inline text does not have a wrapping element, this makes it impossible to use `has` or `+`.

Reply:
Also, this seems like a design choice, so I would like to suggest a Figma tickets first.

Reply:
Also this is not how it works:
`.ams-button:has(.ams-button__icon-start)`
It should be `.ams-button.ams-button__icon-start` because .ams-button__icon-start is not a child of the button. .ams-icon is though.

Reply:
This is the complete picture; I do think it would work. But we decided against changing the padding anyway.

```tsx
return (
  <button … className={clsx('ams-button', …)}>
    {iconPosition === 'start' && <Icon className="ams-button__icon-start" … />}
    …
  </button>
)
```

```html
<button class="ams-button">
  <svg class="ams-icon ams-button__icon-start" />
  Button label
</button>
```

```css
.ams-button:has(ams-button__icon-start) {
  padding-inline-start: …;
}